src/clj/rems/form_validation.clj

```diff
@@ -52,6 +57,23 @@
       {:field-id (:field/id field)
        :type     :t.form.validation/invalid-phone-number})))
 
+(defn- invalid-ip-address-error [field]
+  (when (and (= (:field/type field) :ip-address)
+             (not (str/blank? (:field/value field))))
+    (cond
+      (or
+       (and (first (re-matches +valid-ip-address-regex+ (:field/value field)))
+            (first (re-matches +reserved-ip-address-range-regex+ (:field/value field))))
+       (and (first (re-matches +valid-ip-address-regex-version-six+ (:field/value field)))
+            (first (re-matches +reserved-ip-address-range-regex-version-six+ (:field/value field)))))
+      {:field-id (:field/id field)
+       :type     :t.form.validation/invalid-ip-address-private}
+      (and
+       (not (first (re-matches +valid-ip-address-regex+ (:field/value field))))
+       (not (first (re-matches +valid-ip-address-regex-version-six+ (:field/value field)))))
+      {:field-id (:field/id field)
+       :type     :t.form.validation/invalid-ip-address}
+      :else nil)))
```

Comment:
suggestion: this cond could be simplified a bit like this:

```
(cond
  (first (re-matches +valid-ip-address-regex+ (:field/value field)))
  (when (first (re-matches +reserved-ip-address-range-regex+ (:field/value field)))
    {:field-id (:field/id field)
     :type     :t.form.validation/invalid-ip-address-private})
  
  (first (re-matches +valid-ip-address-regex-version-six+ (:field/value field)))
  (when (first (re-matches +reserved-ip-address-range-regex-version-six+ (:field/value field)))
    {:field-id (:field/id field)
     :type     :t.form.validation/invalid-ip-address-private})
  
  :else ; not a valid ipv4 or ipv6 regex
  {:field-id (:field/id field)
   :type     :t.form.validation/invalid-ip-address})

Reply:
If you do not mind, Ill leave it the way it is